reason.html.markdown

```diff
@@ -0,0 +1,545 @@
+  };
+
+let evan = {firstName: "Evan", age: 14};
+
+/* Pattern match on the exception Under_Age */
+switch (driveToTown(evan)) {
+| status => print_endline(status)
+| exception Under_Age =>
+  print_endline(evan.firstName ++ " is too young to drive!")
+};
+
+/* Alternatively, a try block can be used */
+let messageToEvan =
+  try (driveToTown(evan)) {
+  | Under_Age => evan.firstName ++ " is too young to drive!"
+  };
+/* - With Reason exceptions can be avoided with optionals and are seldom used */
+
+/*----------------------------------------------
+ * Object
+ *----------------------------------------------
+ * Objects are similar to Record types but aren't as rigid
+ * An object resembles a class
+ */
+
+/* An object may be typed like a record but contains a dot */
+type surfaceComputer = {
+  .
+  color: string,
+  capacity: int,
+};
+/* - A single dot signifies a closed object, an object that uses this type
+   must have the exact shape */
+
+let surfaceBook: surfaceComputer = {pub color = "blue"; pub capacity = 512};
+
+/* But an object doesn't require a type */
+let house = {
+  /* A private property */
+  val temp = ref(18.0);
+  /* Public properties */
+  pub temperature = temp;
+  /* A private method only accessible from within house */
+  pri setThermostat = v => temp := v;
+  /* A public method that calls the private setThermostat method */
+  pub arriveHome = () => this#setThermostat(22.0)
+};
+
+house#temperature; /* - : float = 18. */
+house#arriveHome();
+house#temperature; /* - : float = 22. */
+
+/*----------------------------------------------
+ * Module
+ *----------------------------------------------
+ * Modules are used to organize your code and provide namespacing,
+ * Each file is a module by default
+ */
+
+/* Create a module */
+module Staff = {
+  type role =
+    | Delivery
+    | Sales
+    | Other;
+  type member = {
+    name: string,
+    role,
+  };
+
+  let getRoleDirectionMessage = staff =>
+    switch (staff.role) {
+    | Delivery => "Deliver it like you mean it!"
+    | Sales => "Sell it like only you can!"
+    | Other => "You're an important part of the team!"
+    };
+};
+
+/* A module can be accessed with dot notation */
+let newEmployee: Staff.member = {name: "Laura", role: Staff.Delivery};
+
+/* Using the module name can be tiresome so the module's contents can be opened
+   into the current scope with `open` */
+open Staff;
+
+let otherNewEmployee: member = {name: "Fred", role: Other};
+
+/* A module can be extended using the `include` keyword, include copies
+   the contents of the module into the scope of the new module */
+module SpecializedStaff = {
+  include Staff;
+
+  /* `member` is included so there's no need to reference it explicitly */
+  let ceo: member = {name: "Reggie", role: Other};
+
+  let getMeetingTime = staff =>
+    switch (staff) {
+    | Other => 11_15 /* - : int = 1115; Underscores are for formatting only  */
+    | _ => 9_30
+    };
+};
+```
+
+## Further Reading
```

Comment:
Thank you for writing all of this. I'm a big fan of Reason.

Reply:
No problem, I had fun doing it.